omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/AdministrationLinksResource2_0.java

```diff
@@ -0,0 +1,184 @@
+			description.addProperty(DISPLAY);
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty(UUID);
+			description.addProperty(DISPLAY);
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty(UUID);
+			description.addProperty(DISPLAY);
+			description.addProperty(MODULE_TITLE);
+			description.addProperty(LINKS);
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+
+	@PropertyGetter(UUID)
+	public static String getUuid(AdministrationSectionLinks instance) {
+		return instance.getModuleId();
+	}
+
+	@PropertyGetter(DISPLAY)
+	public static String getDisplay(AdministrationSectionLinks instance) {
+		return instance.getTitle();
+	}
+
+	@PropertyGetter(LINKS)
+	public static Map<String, String> getLinks(AdministrationSectionLinks instance) {
+		return instance.getLinks();
+	}
+
+	@Override
+	public Model getGETModel(Representation rep) {
+		ModelImpl model = (ModelImpl) super.getGETModel(rep);
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			model
+					.property(MODULE_TITLE, new StringProperty())
+					.property(LINKS, new ArrayProperty(new ObjectProperty()));
+		}
+
+		return model;
+	}
+
+	@Override
+	public NeedsPaging<AdministrationSectionLinks> doGetAll(RequestContext context) throws ResponseException {
+		return new NeedsPaging<>(getAllAdministrationLinks(), context);
+	}
+
+	private AdministrationSectionLinks getAdministrationLinksForModule(String moduleId) {
+		List<Extension> adminListsExtensions = moduleFactoryWrapper.getExtensions(ADMIN_LIST_POINT_ID);
+		for (Extension adminListExtension : adminListsExtensions) {
+			if (adminListExtension instanceof AdministrationSectionExt && adminListExtension.getModuleId()
+					.equals(moduleId)) {
+				return mapAdminListExtension(adminListExtension);
+			}
+		}
+
+		return null;
+	}
+
+	private List<AdministrationSectionLinks> getAllAdministrationLinks() {
+		List<AdministrationSectionLinks> modulesWithLinksList = new ArrayList<>();
+
+		List<Extension> adminListsExtensions = moduleFactoryWrapper.getExtensions(ADMIN_LIST_POINT_ID);
+
+		for (Extension adminListExtension : adminListsExtensions) {
+			if (adminListExtension instanceof AdministrationSectionExt) {
+				modulesWithLinksList.add(mapAdminListExtension(adminListExtension));
+			}
+		}
+
+		return modulesWithLinksList;
+	}
+
+	private AdministrationSectionLinks mapAdminListExtension(Extension extension) {
+		MessageSourceService messageSourceService = Context.getMessageSourceService();
```

Comment:
Should we get the message source service again each time this method is called?

Reply:
We shouldn't :) Fixed in https://github.com/openmrs/openmrs-module-webservices.rest/pull/484/commits/c78f4e6a3cdc814afb6d993105797912dae6e61c